src/main/java/htsjdk/tribble/readers/AsciiLineReader.java

```diff
@@ -36,18 +35,36 @@
  * @author jrobinso
  */
 public class AsciiLineReader implements LineReader, LocationAware {
+    private static final Log log = Log.getInstance(AsciiLineReader.class);
+
     private static final int BUFFER_OVERFLOW_INCREASE_FACTOR = 2;
     private static final byte LINEFEED = (byte) ('\n' & 0xff);
     private static final byte CARRIAGE_RETURN = (byte) ('\r' & 0xff);
 
-    private final PositionalBufferedStream is;
+    private PositionalBufferedStream is;
     private char[] lineBuffer;
     private int lineTerminatorLength = -1;
 
+    protected AsciiLineReader() {};
+
+    /**
+     * TODO: This class implements LocationAware, which requires preservation of virtual file pointers on BGZF inputs.
+     * However, if the inputStream wrapped by this class is a BlockCompressedInputStream, it violates that contract by
+     * wrapping the stream and returning positional file offsets instead.
+     *
+     * @deprecated 8/8/2017 see {@link #from}
```

Comment:
We've started adding the dates so we can get a rough sense of how long something has been deprecated.  We could always look in the blame which is more accurate, but it's a pain if you're scanning through lots of things.  We could provide the version number, but our deprecation removal policy (which doesn't really exist in any official form) is "has this been deprecated for a long time" rather than "has this been deprecated for N versions".